src/react-components/ory/helpers/filter-flow-nodes.tsx

```diff
@@ -25,20 +25,28 @@ export const FilterFlowNodes = ({
 
   const nodes = filterNodesByGroups(filter)
     // we don't want to map the csrf token every time, only on the form level
-    .filter((node) =>
-      getInputName(node) === "csrf_token" && !includeCSRF ? false : true,
-    )
-    .map((node, k) =>
-      ["hidden"].includes(getNodeInputType(node.attributes))
-        ? {
-            node: <Node node={node} key={k} {...overrides} />,
-            hidden: true,
+    .filter((node) => includeCSRF || !(getInputName(node) === "csrf_token"))
+    .map((node, k) => ({
+      node: (
+        <Node
+          node={node}
+          key={
+            // input node
+            "name" in node.attributes
+              ? node.attributes.name
+              : // image node
+              "src" in node.attributes
+              ? node.attributes.src
+              : // anchor, text & script node
+              "id" in node.attributes
+              ? node.attributes.id
+              : k
```

Comment:
This is the real fix. Previously a re-render after a flow was updated resulted in the same keys, although the input was vastly different. Because the mapping key here was the same, react did not properly re-render the input fields.
This is why index-based keys are always a bit tricky and should only be used when you are absolutely sure that the other props never change across re-renders.

Reply:
makes sense!